### PR TITLE
em-338 submit comment functionality

### PR DESCRIPTION
**Pull request by btzc**
Submit comment functionality finished
- Created validation for form
- Refactored Comment Period code to have partials and it's own module
- Updated tests

### CODE REVIEW

File: src/app/comment-period/submit-comment-modal/submit-comment-modal.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (btzc) [inline src/app/comment-period/submit-comment-modal/submit-comment-modal.ts]: This file is all new. Validation logic, error handling, and comment submission handled here.

File: src/app/services/api.ts [hunk 2]
```diff
@@ -69,11 +69,25 @@ export class Api {
     return this.put(`vclist`, vcsCodes);
   }
 
+  submitDocument(projectId: number, form: FormData, options: Object) {
```
Comment (btzc) [inline src/app/services/api.ts]: Added two new api endpoints, created http post function

File: src/app/services/comment-period.service.ts [hunk 3]
```diff
@@ -8,16 +8,73 @@ import { Observable } from 'rxjs/Observable';
...
 import { ValuedComponent } from '../models/vcs';
 
 @Injectable()
 export class CommentPeriodService {
   pcp: CommentPeriod;
+  comment: Object;
 
   constructor(private api: Api) { }
 
+  // submit a comment
```
Comment (btzc) [inline src/app/services/comment-period.service.ts]: Submit comment service. Depending on whether there's documents or no documents, executes different logic.

File: src/app/comment-period/submit-comment-modal/submit-comment-modal.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    const fileInput = <HTMLInputElement>document.getElementById('file');
+    if (event.target.files.length > 0) {
+      const filesList = event.target.files;
+      for (let i = 0; i < filesList.length; i++ ) {
+        if (this.files.length > 0) {
+          const namesArray = this.files.map(file => file.name);
+          if (!namesArray.includes(filesList[i].name)) {
+            this.files.push(filesList[i]);
+          } else {
+            console.log('error stuff here');
```
Comment (esune) [inline src/app/comment-period/submit-comment-modal/submit-comment-modal.ts]: We should update the error message to be descriptive, or remove it altogether if unnecessary.

**Pull request merged**